scripts/create_eks_cluster.sh

```diff
@@ -0,0 +1,261 @@
+	                k) KEYNAME=${OPTARG};;
+	                t) INSTANCE_TYPE=${OPTARG};;
+	                n) INSTANCE_NUM=${OPTARG};;
+	                f) CALICO_VPP_YAML=${OPTARG};;
+	               \?) echo "Invalid option : -${OPTARG}"; usage;;
+	                :) echo "Option -${OPTARG} requires an argument"; usage;;
+	        esac
+	done
+	NODEGROUP_NAME=$CLUSTER_NAME-nodegroup
+	LT_NAME=$CLUSTER_NAME-lt
+	SSH_SECURITY_GROUP_NAME="$CLUSTER_NAME-ssh-allow"
+fi
+
+### Check for mandatory options
+if [ "$CLUSTER_NAME" = "" -o "$REGION" = "" -o "$KEYNAME" = "" ]; then
+	echo "Please provide \"cluster name\", \"region name\" and \"keyname\"."
+	echo
+	usage
+	exit 1
+fi
+
+### Basic sanity checks; save pain of tons of unnecessary errors and wasted time
+# Make sure we have aws cli version 2
+which aws
+if [ $? -ne 0 ]; then
+	echo "You do not have aws cli installed. Please install aws cli version 2."
+	exit 1
+fi
+AWSCLI_VERSION=`aws --version | awk '{print $1}' | awk -F/ '{print $2}'`
+AWSCLI_VERSION_MAJ=`aws --version | awk '{print $1}' | awk -F/ '{print $2}' | awk -F. '{print $1}'`
+if [ "$AWSCLI_VERSION_MAJ" -ne "2" ]; then
+	echo "You have aws cli version $AWSCLI_VERSION. Please install aws cli version 2."
+	exit 1
+fi
+
+# Make sure we have eksctl version >= 0.51.0; this is the version recommended by aws
+which eksctl
+if [ $? -ne 0 ]; then
+	echo "You do not have eksctl installed. Please install eksctl version >= 0.51.0"
+	exit 1
+fi
+EKSCTL_VERSION=`eksctl version`
+#EKSCTL_VERSION_MAJ=`eksctl version | awk -F. '{print $1}'`
+EKSCTL_VERSION_MIN=`eksctl version | awk -F. '{print $2}'`
+if [  $EKSCTL_VERSION_MIN -lt 51 ]; then
+	echo "You have eksctl version $EKSCTL_VERSION. Please install eksctl version >= 0.51.0"
+	exit 1
+fi
+
+# Make sure we have kubectl
+which kubectl
+if [ $? -ne 0 ]; then
+	echo "You do not have kubectl installed. Please install kubectl."
+	exit 1
+fi
+
+
+### Create a temporary dir
+TMP_DIR=`mktemp -d`
+CUR_DIR=`pwd`
+cd $TMP_DIR
+
+
+### Create cluster without nodegroup
+echo; echo
+echo "Creating cluster $CLUSTER_NAME..."
+echo "NOTE: It will write the config in ~/.kube/eksctl/clusters/$CLUSTER_NAME"
+echo "Please execute \"export KUBECONFIG=~/.kube/eksctl/clusters/$CLUSTER_NAME\" before executing kubectl commands on this cluster"
+echo
+eksctl create cluster --name=$CLUSTER_NAME --without-nodegroup --auto-kubeconfig
+if [ $? -ne 0 ]; then
+	echo
+	echo "ERROR: Failed to create cluster. Refer to the error logs above."
+	cd $CUR_DIR; rm -rf $TMP_DIR; exit 1
+fi
+export KUBECONFIG=~/.kube/eksctl/clusters/$CLUSTER_NAME
+
+### Delete aws-node
+echo; echo
+echo "Deleting aws-node..."
+kubectl delete daemonset -n kube-system aws-node
+
+### Grab the clusterSecurityGroup and VpcId
+CLUSTER_SECURITY_GROUP_ID=`aws eks describe-cluster --name $CLUSTER_NAME | grep RESOURCESVPCCONFIG | awk '{print $2}'`
+if [ "$CLUSTER_SECURITY_GROUP_ID" = "" ]; then
+	echo "ERROR: Missing clusterSecurityGroup. Exiting..."
```

Comment:
Just tested the script, we should force the output format for the aws CLIs, otherwise this section can fail (on my system aws outputs json by default, which breaks the grep / awk).

Reply:
Fixed!

Reply:
I always specify "text" when running "aws configure" so never saw this issue. Thank you for catching this and for the fix :)